graylog2-web-interface/src/views/components/sidebar/NavItem.jsx

```diff
@@ -1,65 +1,130 @@
 // @flow strict
 import * as React from 'react';
 import PropTypes from 'prop-types';
-// $FlowFixMe: could not find types
 import { SizeMe } from 'react-sizeme';
+import styled, { css } from 'styled-components';
 
-import styles from './SideBar.css';
+import { Icon } from 'components/common';
 import CustomPropTypes from '../CustomPropTypes';
 
+type StyleProps = {
+  isSelected: boolean,
+  expandRight: boolean,
+};
+
+export const Title: React.ComponentType<StyleProps> = styled.div`
+  padding: 9px 10px;
+  display: flex;
+  align-items: center;
+  cursor: pointer;
+  position: relative;
+
+  ${props => (props.isSelected ? css`
+    color: #FF3633;
+    background: #393939;
+  ` : '')}
```

Comment:
I could not figure out a way to use something like
```
${props => (props.isSelected && css`
  color: #FF3633;
  background: #393939;
`)}
```


It throws the following flow error:
Cannot assign template string to `Title` because  `InterpolatableComponent` [1] is incompatible with  `React.ComponentType`

Reply:
You could try something like
```
color: ${({ isSelected }) => isSelected ? '#FF3633' : 'currentcolor';
background-color: ${({ isSelected }) => isSelected ? '#393939' : 'currentcolor';
```

or maybe without the `css` function
```
${props => (props.isSelected && `
  color: #FF3633;
  background: #393939;
`)}
```

Reply:
You could also do something like

```
export const Title: React.ComponentType<StyleProps> = styled.div(props => `
/* ... */
color: ${props.isSelected ? '#FF3633' : 'currentcolor'};
/* ... */
`);
```

Reply:
Thanks for the hints! I did not thought about `styled.div({ isSelected } => css`. It allows the usage of 
```
${(isSelected && css`
    color: #FF3633;
    background: #393939;
`)}
```
This will be very helpful in the future! But in this case I will use something like your first example.

Reply:
Just saw I still get a different Flow error. I am trying to find a solution.

Reply:
In doubt I would check if our flow typed definitions are uptodate?